SlabReinforcement/SlabReinforcement.py

```diff
@@ -0,0 +1,363 @@
+# -*- coding: utf-8 -*-
+# ***************************************************************************
+# *                                                                         *
+# *   Copyright (c) 2021 - Shiv Charan <shivcharanmt@gmail.com>             *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "Slab Reinforcement"
+__author__ = "Shiv Charan"
+__url__ = "https://www.freecadweb.org"
+
+import FreeCAD
+from Rebarfunc import (
+    getFacenamesforBeamReinforcement,
+    getParametersOfFace,
+    get_rebar_amount_from_spacing,
+)
+from StraightRebar import makeStraightRebar
+from UShapeRebar import makeUShapeRebar
+from BentShapeRebar import makeBentShapeRebar
+from LShapeRebar import makeLShapeRebar
+from RebarData import RebarTypes
+
+
+def makeSlabReinforcement(
+    rebar_type,
+    front_cover,
+    rear_cover,
+    left_cover,
+    right_cover,
+    top_cover,
+    bottom_cover,
+    diameter,
+    amount_spacing_check,
+    amount_spacing_value,
+    rounding=2,
+    mesh_cover_along="Bottom",
+    structure=None,
+    facename=None,
+    bent_bar_length=50,
+    bent_bar_angle=135,
+):
```

Comment:
you `typing` annotation for arguments. 

https://docs.python.org/3/library/typing.html

Reply:
Added typing annotations, here is commit: https://github.com/amrit3701/FreeCAD-Reinforcement/pull/156/commits/5357f6de322e4c25b57e7b47660b1e125b330e89

Reply:
nice